flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalog.java

```diff
@@ -494,8 +495,22 @@ private static CatalogBaseTable instantiateCatalogTable(Table hiveTable) {
 		String comment = properties.remove(HiveCatalogConfig.COMMENT);
 
 		// Table schema
+		List<FieldSchema> fields;
+		if (org.apache.hadoop.hive.ql.metadata.Table.hasMetastoreBasedSchema(hiveConf,
+				hiveTable.getSd().getSerdeInfo().getSerializationLib())) {
```

Comment:
It seems that Table.getCols() is already doing what it's done here. Could you please check?

Reply:
I think you mean `org.apache.hadoop.hive.ql.metadata.Table.getCols()`. But we only have an instance of `org.apache.hadoop.hive.metastore.api.Table` here. Do you think we should create a `org.apache.hadoop.hive.ql.metadata.Table` and call `getCols()`?

Reply:
Unfortunately we cannot use `Table.getCols()` because this method gets HiveConf from SessionState, which we don't possess on Flink side.